docs/en/guides/pipeline-integrations/development/sgtk-developer-generating-path-and-publish.md

```diff
@@ -0,0 +1,567 @@
+
+You need to ensure that the folders exist on disk so that you can resolve the path in a later step.
+You will use the [Sgtk.create_filesystem_structure()](https://developer.shotgunsoftware.com/tk-core/core.html?#sgtk.Sgtk.create_filesystem_structure) method achieve this:
+
+```python
+tk.create_filesystem_structure("Task", context.task["id"])
+```
+You can use the context object to get the task id to generate the folders.
+
+Your code should now look like this:
+
+```python
+import sgtk
+
+# Get the engine instance that is currently running.
+current_engine = sgtk.platform.current_engine()
+
+# Grab the pre-created Sgtk instance from the current engine.
+tk = current_engine.sgtk
+
+# Get a context object from a Task, this Task must belong to a Shot for the future steps to work. 
+context = tk.context_from_entity("Task", 13155)
+
+# Create the required folders based upon the task.
+tk.create_filesystem_structure("Task", context.task["id"])
+```
+
+You've now completed all the preparation steps and are ready to move onto generating a path using a template.
+
+## Part 5 - Using a template to build a path
+
+### Generating the path
+
+Whenever you need to know where a file should be placed or found in Toolkit you can use the templates to resolve an absolute path on disk.
+
+The first thing you need to do is get a template instance for the path you wish to generate.
+
+```python
+template = tk.templates["maya_shot_publish"]
+```
+
+In this example, you will use the `maya_shot_publish` template. 
+In the default configuration the unresolved template path looks like this:
+
+```yaml
+'sequences/{Sequence}/{Shot}/{Step}/work/maya/{name}.v{version}.{maya_extension}'
+```
+
+The template is made up of keys that you will need to resolve into actual values.
+Since the context contains enough information for the majority of the keys, you can start by using that to extract values:
+
+```python
+fields = context.as_template_fields(template)
+
+>> {'Sequence': 'seq01_chase', 'Shot': 'shot01_running_away', 'Step': 'comp'}
+```
+The [`Context.as_template_fields()`](https://developer.shotgunsoftware.com/tk-core/core.html#sgtk.Context.as_template_fields) gives you a dictionary with the correct values to resolve the template keys. 
+However, it's not provided values for all the keys. The `name`, `version` and `maya_extension` are still missing.
+
+The `maya_extension` key [defines a default value](https://github.com/shotgunsoftware/tk-config-default2/blob/v1.2.8/core/templates.yml#L139) in the template keys section
+so you don't need to provide a value for that, although you could if you wanted a value other than the default.
+
+Which leaves the `name` and `version`. Since the name is a matter of choice, you can either hard code a default or pop up an interface asking the user to provide a value.
+For now, you will hard code both, but in the next step, we'll cover how to find the next available version number.
+
```

Comment:
good point, I've gone with:

> Since the name is a matter of choice, you can either hard code a default or provide an opportunity for the user to enter a value, for example, by popping up an interface.